java/trace/src/main/scala/org/typelevel/otel4s/java/trace/TraceScope.scala

```diff
@@ -78,13 +77,16 @@ private[java] object TraceScope {
               createScope(Scope.Noop)
           }
 
-        def noopScope: Resource[F, Unit] =
+        def noopScope: F ~> F =
           createScope(Scope.Noop)
 
-        private def createScope(scope: Scope): Resource[F, Unit] =
-          Resource
-            .make(local.getAndSet(scope).to[F])(p => local.set(p).to[F])
-            .void
+        private def createScope(scope: Scope): F ~> F =
```

Comment:
This is a signature we can make work with any `cats.mtl.Local`, and makes the whole thing go.